File: backend/schemas/meeting.py
from pydantic import ConfigDict, BaseModel
from datetime import datetime
from typing import Optional

class MeetingCreate(BaseModel):
    title: str

class MeetingResponse(BaseModel):
    id: str
    title: str
    status: str
    job_id: Optional[str] = None
    created_at: datetime
    folder_id: Optional[str] = None
    duration_minutes: Optional[str] = None
    language: Optional[str] = None
    provider: Optional[str] = None
    model_used: Optional[str] = None
    model_version: Optional[str] = None
    processing_time: Optional[str] = None
    word_count: Optional[str] = None
    speaker_count: Optional[str] = None

    model_config = ConfigDict(from_attributes=True, protected_namespaces=())

class MeetingUpdate(BaseModel):
    title: Optional[str] = None
    folder_id: Optional[str] = None

class ChatMessage(BaseModel):
    role: str
    content: str

class ChatRequest(BaseModel):
    messages: list[ChatMessage]

File: shared/types/meeting.ts


export interface Meeting {
  id: string;
  title: string;
  status: 'RECORDING' | 'TRANSCRIBING' | 'SUMMARIZING' | 'EXTRACTING_ACTIONS' | 'DRAFTING_EMAIL' | 'COMPLETED' | 'FAILED';
  created_at: string;
  folder_id?: string | null;
  duration_minutes?: number | null;
  language?: string | null;
  provider?: string | null;
  model_used?: string | null;
  model_version?: string | null;
  processing_time?: number | null;
  word_count?: number | null;
  speaker_count?: number | null;
}

export interface Folder {
  id: string;
  name: string;
  created_at: string;
}
